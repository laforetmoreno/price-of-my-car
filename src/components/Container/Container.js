import React from 'react';
import { node, bool, string } from 'prop-types';

import style from './Container.module.scss';

const Container = ({ children, flex, direction, align, justify }) => (
  <main
    className={`
    ${style.wrapper}
    ${flex && style.flex}
    ${direction && style[direction]}
    ${align && style[align]}
    ${justify && style[justify]}`}
  >
    {children}
  </main>
);

Container.propTypes = {
  children: node.isRequired,
  flex: bool,
  direction: string,
  align: string,
  justify: string,
};

export default Container;
