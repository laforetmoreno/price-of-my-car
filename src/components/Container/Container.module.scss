.wrapper {
  height: calc(100vh - 30px);
  padding: 15px;
  margin-right: auto;
  margin-left: auto;

  @media (min-width: 768px) {
    width: 750px;
  }

  @media (min-width: 992px) {
    width: 970px;
  }

  @media (min-width: 1200px) {
    width: 1170px;
  }
}

.flex {
  display: flex;
}

.column {
  flex-direction: column;
}

.center {
  justify-content: center;
  align-items: center;
}

.between {
  justify-content: space-between;
}

.end {
  align-items: flex-end;
}
