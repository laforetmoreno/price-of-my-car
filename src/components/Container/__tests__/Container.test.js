import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../Container';

describe('<Container />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Container>
          <h1>test</h1>
        </Container>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
