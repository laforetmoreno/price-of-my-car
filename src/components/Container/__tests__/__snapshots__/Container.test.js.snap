// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> renders correctly 1`] = `
<main
  className="wrapper"
>
  <h1>
    test
  </h1>
</main>
`;
